parsl/dataflow/dflow.py

```diff
@@ -920,10 +921,14 @@ def wait_for_current_tasks(self):
             if task_id not in self.tasks:
                 logger.debug("Task {} no longer in task list".format(task_id))
             else:
-                fut = self.tasks[task_id]['app_fu']
+                task_record = self.tasks[task_id]  # still a race condition with the above self.tasks if-statement
```

Comment:
Is there a race-condition you've observed here, can you clarify?

Reply:
The race condition is when a task completes and is deleted from the task list in between checking that it is in the task list (the `if` statement) and assuming that it is in the task list (`task_record =`). I haven't encountered it in practice, so I just left a note in case someone does experience it later.